src/vector.c

```diff
@@ -54,18 +54,21 @@ int git_vector_dup(git_vector *v, const git_vector *src, git_vector_cmp cmp)
 
 	assert(v && src);
 
-	GITERR_CHECK_ALLOC_MULTIPLY(&bytes, src->length, sizeof(void *));
-
-	v->_alloc_size = src->length;
+	v->_alloc_size = 0;
+	v->contents = NULL;
 	v->_cmp = cmp ? cmp : src->_cmp;
 	v->length = src->length;
 	v->flags  = src->flags;
 	if (cmp != src->_cmp)
 		git_vector_set_sorted(v, 0);
-	v->contents = git__malloc(bytes);
-	GITERR_CHECK_ALLOC(v->contents);
 
-	memcpy(v->contents, src->contents, bytes);
+	GITERR_CHECK_ALLOC_MULTIPLY(&bytes, src->length, sizeof(void *));
+	if (bytes != 0) {
```

Comment:
```
if (src->length) {
    size_t bytes;
    GITERR_CHECKALLOC_MULTIPLY(&bytes, src->length, sizeof(void *));
    ...
}
```

No need to do unnecessary work in case where `src->length` is already `0`. I think like this intentions also become a bit clearer.